api/src/main/java/io/smallrye/stork/api/ServiceInstance.java

```diff
@@ -50,11 +50,27 @@ default boolean gatherStatistics() {
     }
 
     /**
-     * When {@code gatherStatistics} is enabled, reports the completion of an operation using this service instance.
+     * When {@code gatherStatistics} is enabled, reports the start of an operation using this service instance.
+     */
+    default void recordStart() {
+    }
+
+    /**
+     * When {@code gatherStatistics} is enabled, reports the end of an operation using this service instance.
+     *
+     * <b>Must</b> be called for each call for which {@link #recordStart()} was called, regardless of recording errors
+     */
+    default void recordEnd() {
```

Comment:
I'm not sure to understand the model.
Basically, in case of success, I got end called. In case of failure, I got error and end called. So, when error is called, I need to store the exception and the processing I need to do to handle that error need to happen in the end method.